File: Server/src/main/java/com/webdev/jobify/model/Job.java
package com.webdev.jobify.model;


import com.fasterxml.jackson.annotation.JsonFormat;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;


class TypeOfEmployment {
    public static final int FULL_TIME = 0;
    public static final int PART_TIME = 1;
    public static final int INTERNSHIP = 2;
    public static final int VOLUNTEER = 3;
}

@Entity
@Table(name = "job")
public class Job implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false)
    private String position;

    @Column(nullable = false)
    private String company;

    private String location;

    private int type;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date startDate;

    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date endDate;

    @Column(nullable = false)
    private boolean hidden;

    @ManyToOne
    private Employee employee;

    public Job() {}

    public Job(Long id, String position, String company, int type, Date startDate, Date endDate, boolean hidden, Employee employee) {
        this.id = id;
        this.position = position;
        this.company = company;
        this.type = type;
        this.startDate = startDate;
        this.endDate = endDate;
        this.hidden = hidden;
        this.employee = employee;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public boolean isHidden() {
        return hidden;
    }

    public void setHidden(boolean hidden) {
        this.hidden = hidden;
    }

    public Employee getEmployee() {
        return employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }

    @Override
    public String toString() {
        return "Job{" +
                "id=" + id +
                ", position='" + position + '\'' +
                ", company='" + company + '\'' +
                ", location='" + location + '\'' +
                ", type=" + type +
                ", startDate=" + startDate +
                ", endDate=" + endDate +
                ", hidden=" + hidden +
                ", employee=" + employee.getId() +
                '}';
    }
}

File: Client/src/app/_models/job/job.ts
import * as globals from '../../globals'

export class Job {

    id: number;
    position: string;
    company: string;
    location: string;
    type: TypeOfEmployement;
    typeString: string;
    startDate: string;
    endDate: string;
    hidden: boolean;

    constructor(id: number, position: string, company: string, location: string, type: TypeOfEmployement,
        startDate: string, endDate: string, hidden: boolean) {
        this.id = id;
        this.position = position;
        this.company = company;
        this.location = location;
        this.type = type;
        this.startDate = startDate;
        this.endDate = endDate;
        this.hidden = hidden;
    }

    public beautify() {

      if(this.type === TypeOfEmployement.FULL_TIME){
        this.typeString = 'Full-time';
      }
      else if(this.type === TypeOfEmployement.PART_TIME){
        this.typeString = 'Part-time';
      }
      else if(this.type === TypeOfEmployement.INTERNSHIP){
        this.typeString = 'Internship';
      }
      else if(this.type === TypeOfEmployement.VOLUNTEER){
        this.typeString = 'Volunteer';
      }
      
      if(this.startDate != null) {
        let month = this.startDate.split('-')[1];
        let year = this.startDate.split('-')[0];
        this.startDate = ' ' + globals.findMonth(month) + ' ' + year;
      }
      
      if(this.endDate != null) {
        let month = this.endDate.split('-')[1];
        let year = this.endDate.split('-')[0];
        this.endDate = ' - ' + globals.findMonth(month) + ' ' + year;
      }
    }

}

export enum TypeOfEmployement {
  FULL_TIME = 0,
  PART_TIME = 1,
  INTERNSHIP = 2,
  VOLUNTEER = 3
}
